### ISSUE TITLE
range().rev() produces "arithmetic operation overflowed"

### ISSUE THREAD

**Issue by seppo0010** [JP-Ellis/rust-skiplist#2]
I tried reversing the range function parameters and got the same result.

Example repro case:
```
+    fn range_rev() {
+        let size = 200;
+        let sl: SkipList<_> = (0..size).collect();
+
+        for i in 0..size {
+            for j in i..size {
+                let mut values = sl.range(Included(i), Included(j)).rev().map(|&i| i);
+                let mut expects = range_inclusive(i, j);
+
+                for (v, e) in values.by_ref().zip(expects.by_ref()) {
+                    assert_eq!(v, e);
+                }
+                assert_eq!(values.next(), None);
+                assert_eq!(expects.next(), None);
+            }
+        }
+    }
+
+    #[test]
```

**Comment by JP-Ellis** [comment]
Similarly, [this works with `BTreeMap`](https://play.rust-lang.org/?gist=56b051d59404b0b67ca5&version=nightly), so I shall fix this too.  Thanks for the report!

**Comment by JP-Ellis** [comment]
Fixed in ae16a74e7340abef675f13de58f1b507246dbeab.
